### PR TITLE
Add the RubLight object

### PR DESCRIPTION
**Pull request by HybridDog**
It glows when it is rubbed, e.g. when Tux or a badguy walks on it.
Skidding makes it glow bright.

I think this object could be used to fix the dark crystal mine issue #1317.
fot_2020-06-03-20:05:27_545027911

I have never added a new object to SuperTux, so please tell me if I did something wrong.

It has not much to do with [that Rublight](https://www.youtube.com/watch?v=bGROryzHLII). I've given it this name because [Triboluminescence](https://en.wikipedia.org/wiki/Triboluminescence) literally means rublight and Triboluminescence is probably more complicated and realistic than what I have implemented.

### CODE REVIEW

File: src/badguy/walking_badguy.hpp [hunk 1]
```diff
@@ -53,7 +53,8 @@ class WalkingBadguy : public BadGuy
 
   void active_update(float dt_sec, float target_velocity);
 
-  float get_velocity_y() const;
+  float get_velocity_y() const { return m_physic.get_velocity_y(); }
```
Comment (tobbi) [inline src/badguy/walking_badguy.hpp:56]: Please re-order these so x comes before y (I know, I'm too OCD about this).

File: src/object/rublight.cpp [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    if (get_brightness() < 0.000001f) {
+      stored_energy = 0;
+      state = STATE_DARK;
+    }
+    break;
+  }
+}
+
+float
+RubLight::get_brightness()
```
Comment (tobbi) [inline src/object/rublight.cpp:97]: Mark this method `const`.

**Comment by weluvgoatz** [comment]
Is the color of the light here configurable?  If not, it would be nice to have.

**Comment by HybridDog** [comment]
> Is the color of the light here configurable? If not, it would be nice to have.

I've made the color and fading speed configurable.
The default value of the `fading_speed` appears in three places: the constructor, in the `mapping.get` invocation and in the `get_settings` method. I've looked at the spotlight code for how to do it; maybe some of them are redundant.

**Comment by tobbi** [comment]
One small thing from Travis:
```
/rublight.cpp.o -c /Users/travis/build/SuperTux/supertux/src/object/rublight.cpp

/Users/travis/build/SuperTux/supertux/src/object/rublight.cpp:34:3: error: field 'fading_speed' will be initialized after field 'light' [-Werror,-Wreorder]

  fading_speed(5.0f),

  ^

1 error generated.
```

**Comment by HybridDog** [comment]
> One small thing from Travis:

it should be fixed now

**Pull request merged**